xbmc/pvr/windows/GUIWindowPVRGuide.cpp

```diff
@@ -695,13 +698,37 @@ CPVRRefreshTimelineItemsThread::CPVRRefreshTimelineItemsThread(CGUIWindowPVRGuid
 
 void CPVRRefreshTimelineItemsThread::Process()
 {
+  // we want this number of refresh calls without new data in a row, before going to normal refresh cycle.
+  static const int BOOSTED_CYCLES_THRESHOLD = 5;
+
+  int iLastEpgItemsCount = 0;
+  int iUpdatesWithoutChange = BOOSTED_CYCLES_THRESHOLD;
+
   while (!m_bStop)
   {
     if (m_pGuideWindow->RefreshTimelineItems() && !m_bStop)
     {
       CGUIMessage m(GUI_MSG_REFRESH_LIST, m_pGuideWindow->GetID(), 0, ObservableMessageEpg);
       KODI::MESSAGING::CApplicationMessenger::GetInstance().SendGUIMessage(m);
     }
-    Sleep(5000);
+
+    // in order to fill the guide window asap, use a short update interval until
+    // BOOSTED_CYCLES_THRESHOLD times in a row we get no more epg events between two cycles.
+    if (iUpdatesWithoutChange)
+    {
+      int iCurrentEpgItemsCount = m_pGuideWindow->CurrentDirectory().Size();
+
+      if (iCurrentEpgItemsCount == iLastEpgItemsCount)
+        iUpdatesWithoutChange--;
+      else
+        iUpdatesWithoutChange = BOOSTED_CYCLES_THRESHOLD; // reset; we want BOOSTED_CYCLES_THRESHOLD in a row
+
+      iLastEpgItemsCount = iCurrentEpgItemsCount;
+    }
+
+    if (iUpdatesWithoutChange)
```

Comment:
@xhaggi just one small change... i wanted 5 calls to refresh without new data in a row, which means 4 times sleep(1000). With your suggested change we would do 5 Sleep(1000). So, the value of the constant needs be changed to 4 now. will change accordingly.